src/FlightDisplay/GuidedAltitudeSlider.qml

```diff
@@ -69,22 +69,27 @@ Rectangle {
             anchors.right:          parent.right
             wrapMode:               Text.WordWrap
             horizontalAlignment:    Text.AlignHCenter
-            text:                   _useAMSL ? qsTr("New Alt(AMSL)") : qsTr("New Alt(rel)")
+            text:                   _useAMSL ? qsTr("New Alt\nAMSL") : qsTr("New Alt\nRel Home")
             color:                  altField.color
         }
 
         QGCLabel {
             id:                         altField
             anchors.horizontalCenter:   parent.horizontalCenter
-            text:                       newAltitudeAppUnits + " " + QGroundControl.unitsConversion.appSettingsHorizontalDistanceUnitsString
-            color:                      !isNaN(_distanceToGround) && (-altLossGain > _distanceToGround) ? qgcPal.warningText : qgcPal.text
+            horizontalAlignment:        Text.AlignHCenter
+            text:                       labelText
+            color:                      !isNaN(_distanceToGround) && ((-altLossGain > _distanceToGround) || (altLossGain > 120.0 - _distanceToGround)) ? qgcPal.warningText : qgcPal.text
 
             property real   altExp:                 Math.pow(altSlider.value, _sliderExponent)
             property real   altLossGain:            altExp * (altSlider.value > 0 ? altGainRange : altLossRange)
             property real   newAltitudeMeters:      _vehicleAltitude + altLossGain
             property real   newAltitudeAMSLMeters:  _vehicleAltitudeAMSL + altLossGain
             property string newAltitudeAppUnits:    QGroundControl.unitsConversion.metersToAppSettingsHorizontalDistanceUnits(
                                                         _useAMSL ? newAltitudeAMSLMeters : newAltitudeMeters).toFixed(1)
+            // Concatenate new altitude and unit with change relative to current altitude                                            
+            property string labelText:              newAltitudeAppUnits + " " + QGroundControl.unitsConversion.appSettingsHorizontalDistanceUnitsString
+                                                        + "\n(" + (altLossGain < 0 ? "" : "+") + QGroundControl.unitsConversion.metersToAppSettingsHorizontalDistanceUnits(altLossGain).toFixed(1)
```

Comment:
+ "\n(" + (altLossGain < 0 ? "" : "+") +
                                                        
                                                        Missing minus sign?

Reply:
The minus sign is implicitly printed, so only (explicitly) need a plus sign.
A better way to do this would be to define a custom format with sign always printed, but this is the fast way.